www/src/components/repository/Repository.tsx

```diff
@@ -23,14 +28,23 @@ import { RepositorySideCar } from './RepositorySideCar'
 
 import { REPOSITORY_QUERY } from './queries'
 
+// To test whether the provided url param is a uuid or a name
+const uuidRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/i
+
 function Repository() {
   const { name } = useParams()
   const [searchParams] = useSearchParams()
-  const { data } = useQuery(REPOSITORY_QUERY, { variables: { name } })
+  const variables: any = { name }
+
+  if (name && uuidRegex.test(name)) {
+    variables.repositoryId = name
+  }
+
+  const { data, loading } = useQuery(REPOSITORY_QUERY, { variables })
   const backStackName = searchParams.get('backStackName')
   const tabStateRef = useRef<any>(null)
 
-  if (!data) {
+  if (loading) {
```

Comment:
this can make the ui flash a ton i think

Reply:
No it does not, loading and data are updated in a batch

Reply:
yea but it can mark as loading when the query is refetched for all sorts of reasons (new props sometimes, updates, etc)

Reply:
I am pretty sure it's done it in the past though, and I don't think we should revert it because it has caused issues already unless we're sure it won't flash around